src/conversions/osstr.rs

```diff
@@ -58,7 +58,22 @@ impl ToPyObject for OsStr {
 
 impl FromPyObject<'_> for OsString {
     fn extract(ob: &PyAny) -> PyResult<Self> {
-        let pystring = <PyString as PyTryFrom>::try_from(ob)?; // Cast PyAny to PyString
+        let pystring = match <PyString as PyTryFrom>::try_from(ob) {
+            // Cast PyAny to PyString
+            Ok(s) => s,
+            Err(err) => {
+                let py = ob.py();
+                let pathlib = py.import("pathlib")?;
+                let pathlib_path = pathlib.getattr("Path")?;
+                // FIXME: right way to get the type object of pathlib.Path?
```

Comment:
Can't you "extract" a `PyType` object from `pathlib_path`?

Reply:
```
error[E0277]: the trait bound `PyType: PyClass` is not satisfied
  --> src/conversions/osstr.rs:68:46
   |
68 |                 let path_type = pathlib_path.extract::<PyType>()?;
   |                                              ^^^^^^^ the trait `PyClass` is not implemented for `PyType`
   |
   = note: required because of the requirements on the impl of `conversion::FromPyObject<'_>` for `PyType`


error[E0277]: the trait bound `PyType: Clone` is not satisfied
  --> src/conversions/osstr.rs:68:46
   |
68 |                 let path_type = pathlib_path.extract::<PyType>()?;
   |                                              ^^^^^^^ the trait `Clone` is not implemented for `PyType`
   |
   = note: required because of the requirements on the impl of `conversion::FromPyObject<'_>` for `PyType`
```

Reply:
Need to use `.downcast()` or `.extract::<&PyType>()` - see my comment below!

Reply:
Sorry, should have specified `&PyType`.